renderdoc/driver/gl/wrappers/gl_draw_funcs.cpp

```diff
@@ -4014,6 +4098,11 @@ void WrappedOpenGL::glClearNamedBufferDataEXT(GLuint buffer, GLenum internalform
 {
   CoherentMapImplicitBarrier();
 
+  if(IsBackgroundCapturing(m_State))
+  {
+    GetResourceManager()->MarkDirtyWithWriteReference(BufferRes(GetCtx(), buffer));
```

Comment:
Similar to other cases this needs to happen while active capturing as well, so we can grab its contents if we need to before the GL call itself.